File: SDKDemo/SDKDemo/NativeAdViewController.m
//
//  NativeAdViewController.m
//  SDKDemo
//
//  Created by 马秉尧 on 2019/9/25.
//  Copyright © 2019 新义互联（北京）科技有限公司. All rights reserved.
//

@import AdtalosSDK;

#import "NativeAdViewController.h"

@interface NativeAdViewController ()<AdtalosListener, AdtalosVideoListener>

- (void) onAdRendered;
- (void) onAdImpressionFinished;
- (void) onAdImpressionReceivedError:(NSError *)error;
- (void) onAdLoaded;
- (void) onAdFailedToLoad:(NSError *)error;
- (void) onAdClicked;
- (void) onAdLeftApplication;
- (void) onAdOpened;
- (void) onAdClosed;

- (void) onVideoLoad:(NSDictionary *)metadata;
- (void) onVideoStart;
- (void) onVideoPlay;
- (void) onVideoPause;
- (void) onVideoEnd;
- (void) onVideoVolumeChange:(double)volume muted:(BOOL)muted;
- (void) onVideoTimeUpdate:(double)currentTime duration:(double)duration;
- (void) onVideoError;
- (void) onVideoBreak;

@end

@implementation NativeAdViewController {
    AdtalosAdView *adView;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    CGRect frame = CGRectMakeWithAdtalosNative7to5AdSize(0, 50);
    adView = [[AdtalosAdView alloc] initWithFrame:frame];
    adView.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin;
    [adView loadAd: @"98738D91D3BB241458D3FAE5A5BF7B34"];
    adView.delegate = self;
    adView.videoController.delegate = self;
    [self.view addSubview:adView];
    
    UIButton *closeButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    CGFloat x = ([UIScreen mainScreen].bounds.size.width - 100) / 2;
    CGFloat y = [UIScreen mainScreen].bounds.size.height - 100;
    closeButton.frame = CGRectMake(x, y, 100, 50);
    closeButton.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin;
    [closeButton setTitle:@"关闭" forState:UIControlStateNormal];
    [closeButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [closeButton setBackgroundColor:[UIColor redColor]];
    [closeButton addTarget:self action:@selector(closeButtonClick:) forControlEvents:UIControlEventTouchUpInside];
    [closeButton.layer setMasksToBounds:YES];
    [closeButton.layer setCornerRadius:10.0];
    [self.view addSubview:closeButton];
}

- (void)closeButtonClick:(UIButton *)button {
    [adView.videoController pause];
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void) onAdRendered {
    NSLog(@"Adtalos onAdRendered");
}

- (void) onAdImpressionFinished {
    NSLog(@"Adtalos onAdImpressionFinished");
}

- (void) onAdImpressionReceivedError:(NSError *)error {
    NSLog(@"Adtalos onAdImpressionReceivedError: %@", error);
}

- (void) onAdLoaded {
    NSLog(@"Adtalos onAdLoaded");
}

- (void) onAdFailedToLoad:(NSError *)error {
    NSLog(@"Adtalos onAdFailedToLoad: %@", error);
}

- (void) onAdClicked {
    NSLog(@"Adtalos onAdClicked");

}

- (void) onAdLeftApplication {
    NSLog(@"Adtalos onAdLeftApplication");
}

- (void) onAdOpened {
    NSLog(@"Adtalos onAdOpened");
}

- (void) onAdClosed {
    NSLog(@"Adtalos onAdClosed");
}

- (void) onVideoLoad:(NSDictionary *)metadata {
    NSLog(@"Adtalos onVideoLoad: %@", metadata);
}

- (void) onVideoStart {
    NSLog(@"Adtalos onVideoStart");
}

- (void) onVideoPlay {
    NSLog(@"Adtalos onVideoPlay");
}

- (void) onVideoPause {
    NSLog(@"Adtalos onVideoPause");
}

- (void) onVideoEnd {
    NSLog(@"Adtalos onVideoEnd");
}

- (void) onVideoVolumeChange:(double)volume muted:(BOOL)muted {
    NSLog(@"Adtalos onVideoVolumeChange: %f, %d", volume, muted);

}

- (void) onVideoTimeUpdate:(double)currentTime duration:(double)duration {
    NSLog(@"Adtalos onVideoTimeUpdate: %f, %f", currentTime, duration);
}

- (void) onVideoError {
    NSLog(@"Adtalos onVideoError");
}

- (void) onVideoBreak {
    NSLog(@"Adtalos onVideoBreak");
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    adView = nil;
    // Dispose of any resources that can be recreated.
}

@end

File: SDKSwiftDemo/SDKSwiftDemo/NativeAdViewController.swift
//
//  NativeAdViewController.swift
//  SDKSwiftDemo
//
//  Created by 马秉尧 on 2019/10/18.
//  Copyright © 2019 新义互联（北京）科技有限公司. All rights reserved.
//

import Foundation

import UIKit
import AdtalosSDK

class NativeAdViewController: UIViewController, AdtalosListener, AdtalosVideoListener {

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.white
        let frame: CGRect = CGRectMakeWithAdtalosNative7to5AdSize(0, 50)
        let adView: AdtalosAdView = AdtalosAdView(frame: frame)
        adView.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin]
        adView.loadAd("98738D91D3BB241458D3FAE5A5BF7B34")
        adView.delegate = self
        adView.videoController.delegate = self;
        view.addSubview(adView)

        let closeButton: UIButton = UIButton(type: .roundedRect)
        let x: CGFloat = (UIScreen.main.bounds.size.width - 100) / 2
        let y: CGFloat = (UIScreen.main.bounds.size.height - 100)
        closeButton.frame = CGRect(x: x, y: y, width:100, height:50)
        closeButton.autoresizingMask = [.flexibleWidth, .flexibleHeight, .flexibleLeftMargin, .flexibleRightMargin, .flexibleTopMargin, .flexibleBottomMargin]
        closeButton.setTitle("关闭", for: .normal)
        closeButton.setTitleColor(UIColor.white, for: .normal)
        closeButton.backgroundColor = UIColor.red
        closeButton.addTarget(self, action: #selector(closeButtonClick), for: .touchUpInside)
        closeButton.layer.masksToBounds = true
        closeButton.layer.cornerRadius = 10.0
        view.addSubview(closeButton)
    }

    @objc func closeButtonClick() {
        dismiss(animated: true, completion: nil)
    }
    
    func onAdRendered() {
        NSLog("Adtalos onAdRendered")
    }

    func onAdImpressionFinished() {
        NSLog("Adtalos onAdImpressionFinished")
    }

    func onAdImpressionReceivedError(_ error: Error!) {
        NSLog("Adtalos onAdImpressionReceivedError")
    }
    
    func onAdLoaded() {
        NSLog("Adtalos onAdLoaded")
    }

    func onAdFailed(toLoad error: Error!) {
        NSLog("Adtalos onAdImpressionReceivedError")
    }

    func onAdClicked() {
        NSLog("Adtalos onAdClicked")

    }

    func onAdLeftApplication() {
        NSLog("Adtalos onAdLeftApplication")
    }

    func  onAdOpened() {
        NSLog("Adtalos onAdOpened")
    }

    func onAdClosed() {
        NSLog("Adtalos onAdClosed")
    }
    
    func onVideoLoad(_ metadata: [AnyHashable : Any]!) {
        NSLog("Adtalos onVideoLoad: %@", metadata);
    }

    func onVideoStart() {
        NSLog("Adtalos onVideoStart");
    }

    func onVideoPlay() {
        NSLog("Adtalos onVideoPlay");
    }

    func onVideoPause() {
        NSLog("Adtalos onVideoPause");
    }

    func onVideoEnd() {
        NSLog("Adtalos onVideoEnd");
    }

    func onVideoVolumeChange(_ volume: Double, muted: Bool) {
        NSLog("Adtalos onVideoVolumeChange: %f, %d", volume, muted);
    }

    func onVideoTimeUpdate(_ currentTime: Double, duration: Double) {
        NSLog("Adtalos onVideoTimeUpdate: %f, %f", currentTime, duration);
    }

    func onVideoError() {
        NSLog("Adtalos onVideoError");
    }

    func onVideoBreak() {
        NSLog("Adtalos onVideoBreak");
    }
}
